### PR TITLE
CompAIRR integration

### PR DESCRIPTION
**Pull request by BlueBasilisk**
* Added a new encoder - EditDistanceEncoder, which internally uses CompAIRR to calculate repertoire overlap 
* Added PrecomputedKNN which is directly compatible with DistanceEncoder and EditDistanceEncoder (without having to specify precomputed=true, which may be hard to guess for a user)
* Minor simplifications in DistanceEncoder and compatibility fixes in DesignMatrixExporter

### CODE REVIEW

**Comment by BlueBasilisk** [comment]
added somethng small to the docs now afterwards because I forgot it earlier, but I am done now :)

File: immuneML/encodings/distance_encoding/EditDistanceEncoder.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+from immuneML.data_model.dataset.RepertoireDataset import RepertoireDataset
+from immuneML.data_model.encoded_data.EncodedData import EncodedData
+from immuneML.encodings.DatasetEncoder import DatasetEncoder
+from immuneML.encodings.EncoderParams import EncoderParams
+from immuneML.environment.EnvironmentSettings import EnvironmentSettings
+from immuneML.util.EncoderHelper import EncoderHelper
+from immuneML.util.ParameterValidator import ParameterValidator
+
+
+class EditDistanceEncoder(DatasetEncoder):
```
Comment (pavlovicmilena) [inline immuneML/encodings/distance_encoding/EditDistanceEncoder.py:19]: is there a specific reason why it's called EditDistance, but computes Morisita-Horn? :) is that configurable or should it always be MH?

File: docs/source/installation/install_with_package_manager.rst [hunk 2]
```diff
@@ -61,6 +61,17 @@ To install the DeepRC dependencies, run:
 
   pip install -r requirements_DeepRC.txt --no-dependencies
 
+6. Optionally, if you want to use the :ref:`EditDistanceEncoder`, you have to install the C++ tool `CompAIRR <https://github.com/uio-bmi/compairr>`_,
```
Comment (pavlovicmilena) [inline docs/source/installation/install_with_package_manager.rst:64]: is it possible to have an easier way to install CompAIRR? will it also be a part of the package or it has to be installed separately? :) we should also update the docker file to include CompAIRR :)

**Review by pavlovicmilena — COMMENTED** [review]
Just had a few questions :)

**Pull request closed without merging**